### PR TITLE
Fix clang build

### PR DESCRIPTION
**Pull request by akva2**
- Forward struct as struct, not class (quells a warning)
- Avoid structured bindings in lambda capture list (requires c++-20)

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

**Review by bska — COMMENTED** [review]
Other than a minor API issue in a lambda, this looks good to me.

File: opm/grid/cpgrid/processEclipseFormat.cpp [hunk 1]
```diff
@@ -242,26 +242,36 @@ namespace cpgrid
...
-                        const auto& [ cell_center_top, face_center, area_normal ] = top_cell_info;
                         const auto& cell_center_bottom = std::get<0>(bottom_cell_info);
                         cells_between.push_back(cell_top);
 
                         auto compute_half_trans =
-                            [&face_center, &area_normal](const std::array<double,3>& cell_center,
-                                                         double perm)
+                            [](const std::array<double,3>& cell_center,
+                               const auto& face_center,
+                               const auto& area_normal,
```
[the remark this answers is not in the document]
Comment (akva2) [reply opm/grid/cpgrid/processEclipseFormat.cpp]: agreed and changed

**Comment by akva2** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks a lot for the updates.  This looks good to me now and I'll merge this into the master and 2024.10 release branches.

**Pull request merged**